api/src/wfl/module/covid.clj

```diff
@@ -207,17 +209,55 @@
         (jdbc/execute! tx [(format query queue) id])))
     (throw (ex-info "No snapshots in queue" {:source source}))))
 
+(defn ^:private go-create-snapshot!
+  "Create snapshot in TDR from `dataset` body, `table`
+   `now` string and `row-ids` then write job info as well
+   as rows into `source-details-name` table."
+  [now dataset table row-ids]
+  (let [columns     (-> (datarepo/all-columns dataset table)
+                        (->> (map :name) set)
+                        (conj "datarepo_row_id"))
+        job-id (-> (datarepo/make-snapshot-request dataset columns table row-ids)
+                   (assoc :name now)
+                   (datarepo/create-snapshot-job))]
+    job-id))
+
 ;; Create and add new snapshots to the snapshot queue
-(defn ^:private update-tdr-source [source]
-  (letfn [(find-new-rows       [source now] [])
-          (make-snapshots      [source row-ids])
-          (write-snapshots     [source snapshots])
-          (update-last-checked [now])]
-    (let [now     (OffsetDateTime/now)
-          row-ids (find-new-rows source now)]
-      (when-not (empty? row-ids)
-        (write-snapshots source (make-snapshots source row-ids)))
-      (update-last-checked source now))))
+(defn ^:private update-tdr-source
+  "Check for new data in TDR, create new snapshots, insert
+   resulting job creation ids into database and update the
+   timestamp for next time using transaction TX."
+  [{:keys [id dataset dataset_table table_column_name last_checked details] :as _source}]
+  (letfn [(find-new-rows [now]
+            (-> (datarepo/query-table-between
+                 dataset
+                 dataset_table
+                 table_column_name
+                 [last_checked now]
+                 [:datarepo_row_id])
+                :rows
+                flatten))
+          (make-snapshots [now-obj row-ids]
+            (let [shards (partition-all 500 row-ids)
+                  compact-now (.format now-obj (DateTimeFormatter/ofPattern "YYYYMMDD'T'HHMMSS"))]
```

Comment:
I will be using this too to generate meaningfully unique-enough snapshot reference names.  Can you break out into a utility please?

Reply:
Actually, as I think about it more, I am using the snapshot name as the base of my snapshot reference name.  So if you're already tagging the name with a timestamp, I don't think I need to do the same (it will get long and confusing).

If you can give me a concrete example of what a snapshot name will be under this code, that would help my development!

Reply:
Actually I think you found a bug! The way it works now is `(assoc :name now)` which does not give what I expect… It should be updated to `(update s :name #(str % now))`, and a concrete example would be: 

```clojure
(go-create-snapshot! now (datarepo/dataset "85efdfea-52fb-4698-bee6-eef76104a7f4")
    "sample" ["1" "2" "3"])

; => 

{:contents [{:datasetName "zerotest_partition",
             :mode "byRowId",
             :rowIdSpec {:tables [{:columns #{"vcf" "datarepo_row_id" "id" "vcf_index"},
                                   :rowIds ["1" "2" "3"],
                                   :tableName "sample"}]}}],
 :description "This is a test dataset for Zero that uses date partitioning.",
 :name "zerotest_partition202105133T150557",
 :profileId "XXXXX"}
```

Reply:
Ahhh!  Great, thanks for looking at this closer and for providing the concrete example.

Reply:
Caught something else:
```
YYYYMMDD'T'HHMMSS
```

Note that `MM` is the month and it's used in both the month and minute spots, think what you want instead is:
```
YYYYMMDD'T'HHmmSS
```

Source:
https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html

Reply:
Thanks for catching, based on https://help.sumologic.com/03Send-Data/Sources/04Reference-Information-for-Sources/Timestamps%2C-Time-Zones%2C-Time-Ranges%2C-and-Date-Formats, I also updated `SS` to `ss`